pkg/issuer/acme/issue.go

```diff
@@ -74,38 +78,88 @@ func (a *Acme) obtainCertificate(ctx context.Context, crt *v1alpha1.Certificate)
 		return nil, nil, fmt.Errorf("error getting certificate private key: %s", err.Error())
 	}
 
-	// generate a csr
-	template, err := pki.GenerateCSR(a.issuer, crt)
-	if err != nil {
-		// TODO: this should probably be classed as a permanant failure
-		return nil, nil, err
+	var certSlice [][]byte
+	// StatusReady indicates that the order is ready to be finalized.
+	// Once the order has been finalized, it will transition into the 'valid'
+	// state.
+	// The only way to obtain a certificate from an already 'valid' order is to
+	// call the orders GetCertificate function.
+	// You can see we do this below *iff* certSlice is nil.
+	if order.Status == acmeapi.StatusReady {
```

Comment:
As the comment explains, if the order is 'ready' then we need to finalize it (i.e. submit a CSR)